### ISSUE TITLE
[Feature Request] Skip filecheck/allocate option

### ISSUE THREAD

**Issue by PrototypeZero** [BiglySoftware/BiglyBT#1045]
Often when restarting a seed it takes ages (talking 30+ minutes each) for the program to check and "allocate" completed torrents (seeds) which I know are good and haven't changed.  

I used to think it was the time between the torrent last being seeded and re-started affected whether it was "fast" checked or not, but apparently not as I can stop 10 torrents for 30 seconds and restart them and different ones get "fast" checked some times and not others.

What is the logic that determines this?

Could you therefore put an option to NOT recheck on restart of individual torrents (maybe if its within a certain time from last being active maybe).  Or perhaps if not that, an instant right click "skip recheck/allocate" option for torrents current checking.

**Comment by parg** [comment]
Right-click -> Advanced -> Set resume data complete (skip recheck)

might help. 

For some reason it appears to have lost state information regarding completeness.

**Comment by PrototypeZero** [comment]
Thanks for the info on how to skip, is there anything I could do to prevent loss of this state info or diagnose why it's being lost?

**Comment by parg** [comment]
Are the downloads definitely stuck in a 'checking state' (i.e. can you see their checking status slowly incrementing in the Library view) or are you talking about them being in an 'allocating' state? Obviously once a download is complete there should be no further need for any actual 'allocation' but during that phase it does check that all the files exist and are the expected lengths. If a length has changed (e.g. metadata has been added to something by a player) then this would cause the download to them proceed into a 'checking' state.

**Comment by PrototypeZero** [comment]
Most of the time the files say "Allocating" (*very* occasionally when restarting 1 torrent may flip to checking, pass the check and never d/l anything more).

But if as you say it is only checking the existence and file length of the files then why is it actually reading the whole contents of the files (as if it was checking them)?  I can tell it's doing this via Resource Monitor and other tools and by the fact it takes forever, especially as most of the torrents are 20-400GB in size.

**Comment by parg** [comment]
If it says 'allocating' for a long time for a complete download then it is actually most likely waiting for another download to allocate as that phase is single-threaded. If it doesn't go into 'checking' then it won't be reading the file's contents.

**Comment by PrototypeZero** [comment]
It does the said behaviour even when there is no torrent downloading (only multiple ones seeding).  And its definitely reading the files contents not just checking file existence/size while displaying "allocating".

**Comment by ferdnyc** [comment]
I don't think I've ever even _seen_ an "Allocating" message — I assume that step is only done if you have one of the "Allocate..." options checked in Options > Files ?

I run with both of the "Allocate..." boxes unchecked, "Use sparse files..." checked (but that's recent, and things worked the same without it), and pretty much everything else unchecked until "Re-check pieces when download is done", which is checked (along with both options after it). I also have "Use Fast Resume mode" enabled, a bit lower down.

Configured that way, I can stop and restart torrents instantly, regardless whether they're in the Complete/Seeding or Incomplete/Downloading state. The only time any re-checking gets forced on me is when recovering from a crash (and with Fast Resume, even that's not a _complete_ re-check), or if I've changed the set of files selected for download.

**Comment by ferdnyc** [comment]
One other thought, @PrototypeZero : You're not downloading into a Dropbox folder or anything like that, are you? Having an external monitoring daemon (like a file-syncing client) messing with its files can interfere with BiglyBT's state-management.

**Comment by PrototypeZero** [comment]
No not using Dropbox or similar.

However I seemed of fixed it (some how) now.  

I looked at the options in Files, I hadn't changed anything (I normally have sparse option selected since its introduction).  

There seemed to be some bug regarding the "Perform low resource recheck of of pieces while seeding", it was de-selected (as I always had been on my systems) but for some reason Bigly thought it wasn't, so I  enabling it then disabled it again and this seems to have worked!

Why it was doing it on 2 separate systems I don't know , but it seems to be resolved for now.

Thanks for the replies and I'll post again if I find a way of reproducing it.
